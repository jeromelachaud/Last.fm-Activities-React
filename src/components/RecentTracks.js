import React from 'react';
import './RecentTracks.css';
import RecentTrack from './RecentTrack.js';
import _map from 'lodash/map';
import _isEqual from 'lodash/isEqual';
import {
  fetchRecentTracks
} from '../actions/action-creators';

let RecentTracks = React.createClass ({

  shouldComponentUpdate(nextProps){
    return !_isEqual(nextProps.recentTracks,this.props.recentTracks);
  },

  componentWillMount() {
    fetchRecentTracks();
    // this.updateInterval = setInterval(() => fetchRecentTracks(2), 3000); 
  },

  render() {

    const {
      recentTracks
    } = this.props;

    let RecentTrackElement;
    if (recentTracks) {
      RecentTrackElement = _map(recentTracks, (recentTrack, i) => (
        <RecentTrack
          key={i}
          recentTrack={recentTrack} />
      ));
    }

    return (
    <section
      className="RecentTracks">
      {RecentTrackElement}
    </section>
    );
  }
});

export default RecentTracks;
