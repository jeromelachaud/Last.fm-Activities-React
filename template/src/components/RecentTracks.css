/* RecentTracks */
.RecentTracks{
  width: 30rem;
  margin: 0 auto
}
