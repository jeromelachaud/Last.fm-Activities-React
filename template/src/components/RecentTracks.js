import React from 'react';
import './RecentTracks.css';
import RecentTrack from './RecentTrack.js';
import _map from 'lodash/map';

let RecentTracks = React.createClass ({

  render() {

    const {
      recentTracks
    } = this.props;

    let RecentTrackElement;
    if (recentTracks) {
      RecentTrackElement = _map(recentTracks, (recentTrack, i) => (
        <RecentTrack
          key={i}
          recentTrack={recentTrack} />
      ));
    }

    return (
    <section
      className="RecentTracks">
      {RecentTrackElement}
    </section>
    );
  }
});

export default RecentTracks;
