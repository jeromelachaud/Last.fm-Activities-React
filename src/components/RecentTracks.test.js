import React from 'React';
import {
  shallow
} from 'enzyme';
import toJson from 'enzyme-to-json';
import RecentTracks from './RecentTracks';

const recentTracks = [];

describe('<RecentTracks> component', () => {
  const wrapper = shallow(<RecentTracks recentTracks={recentTracks} />);

  it('should render as expected', () => {
    expect(toJson(wrapper)).toMatchSnapshot();
  });
});
