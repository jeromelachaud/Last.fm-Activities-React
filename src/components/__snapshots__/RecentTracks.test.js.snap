exports[`<RecentTracks> component should render as expected 1`] = `
<section
  className="RecentTracks" />
`;
